model/src/test/scala/aqua/model/topology/TopologySpec.scala

```diff
@@ -53,6 +53,25 @@ class TopologySpec extends AnyFlatSpec with Matchers {
     proc should be(expected)
   }
 
+  "topology resolver" should "simplify a route" in {
```

Comment:
Please provide a more complex test with a `seq` inside the internal `on`, or another `on`.